### DISCUSSION TITLE
Unable to perform simple tests with CSD components?

### DISCUSSION THREAD

**Discussion by fltzr** [cloudscape-design/components#1692] [Q&A] [answered]
Hello!

Have really been enjoying CSD lately. We are wanting to incorporate this in a corporate setting, and I have been been reading over the documentation for implementing unit-tests.

However, I can not figure out why I am unable to run a simple test on these components. I have tried referring to the [documentation](https://cloudscape.design/get-started/testing/introduction/) and tried reproducing the guidance from #1655 -- I made a clean repo provided from Vitest which you can find [here](https://github.com/fltzr/vitest-cloudscape-unit-test).

Whenever I run the unit test, I am given the following error:

```
 ❯ src/components/sample-modal.test.tsx (1)
   ❯ SampleModal (1)
     × open modal and close

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/components/sample-modal.test.tsx > SampleModal > open modal and close
SyntaxError: \8 and \9 are not allowed in strict mode.
 ❯ compile node_modules/nwsapi/src/nwsapi.js:760:17
 ❯ match_collect node_modules/nwsapi/src/nwsapi.js:1339:16
 ❯ Object._matches [as match] node_modules/nwsapi/src/nwsapi.js:1394:35
 ❯ exports.matchesDontThrow node_modules/jsdom/lib/jsdom/living/helpers/selectors.js:29:36
 ❯ matches node_modules/jsdom/lib/jsdom/living/helpers/style-rules.js:172:10
 ❯ node_modules/jsdom/lib/jsdom/living/helpers/style-rules.js:109:18
 ❯ handleSheet node_modules/jsdom/lib/jsdom/living/helpers/style-rules.js:100:13
```

We are also seeing the same error in our larger repository, but no luck when testing on a clean environment.
To try isolating this from our repo, I removed strict mode from my tsconfig as well as from `main.tsx`, but I am still given the same error on the tests.

I am hoping the solution to this is not too obvious, but would appreciate and help/guidance! Thanks so much!

**Comment by just-boris** [comment, marked as answer]
I think the issue is in this line https://github.com/fltzr/vitest-cloudscape-unit-test/blob/4d50a484c4d1542e445634c72cdeb0db96975628/vite.config.ts#L14

`css: true` option can't be used in vitest with our components. Can you try turning it off and test again?

For additional context, this is sort of a known issue: https://github.com/dperini/nwsapi/issues/92

This module is used in JSDOM, which runs tests in your demo

**Reply by fltzr** [reply]
Ah very interesting that did just the trick! Thanks so much, Boris!
